app/src/androidTest/java/com/group4sweng/scranplan/PlannerFragmentTest.java

```diff
@@ -66,6 +66,54 @@ public void checkAddButton() {
                 matches(isDisplayed()));
     }
 
+    //Check breakfast fragment appears and the recipe in it can be added successfully to the meal plan
+    @Test
+    public void AddBreakfast() throws InterruptedException {
+        Log.d(TAG, "Testing add breakfast");
+
+        onView(withText("Meal Planner")).perform(click());
+        onView(withId(0)).perform(longClick());
+
+        Thread.sleep(THREAD_SLEEP_TIME/4);
+
+        onView(withText("hearty pasta soup")).perform(click());
+        onView(withText("Add")).perform(click());
+
+        Thread.sleep(THREAD_SLEEP_TIME/4);
+    }
+
+    //Check lunch fragment appears and the recipe in it can be added successfully to the meal plan
+    @Test
+    public void checkAddLunch() throws InterruptedException {
+        Log.d(TAG, "Testing add lunch");
+
+        onView(withText("Meal Planner")).perform(click());
+        onView(withId(1)).perform(longClick());
+
+        Thread.sleep(THREAD_SLEEP_TIME/4);
+
+        onView(withText("chicken noodle soup")).perform(click());
+        onView(withText("Add")).perform(click());
+
+        Thread.sleep(THREAD_SLEEP_TIME/4);
+    }
+
+    //Check dinner fragment appears and the recipe in it can be added successfully to the meal plan
+    @Test
+    public void checkAddDinner() throws InterruptedException {
+        Log.d(TAG, "Testing add dinner");
+
+        onView(withText("Meal Planner")).perform(click());
+        onView(withId(2)).perform(longClick());
+
+        Thread.sleep(THREAD_SLEEP_TIME/4);
+
+        onView(withText("slow cooker spiced root and lentil casserole")).perform(click());
+        onView(withText("Add")).perform(click());
+
+        Thread.sleep(THREAD_SLEEP_TIME/4);
+    }
+
```

Comment:
Also need checks for the contents of what is retrieved when you hold press on a recipe. Hence why the issues I presented have surfaced.

Reply:
Also dosen't acctually check anything. Just performs an automated task. Look into using 'assert' conditions from JUnit or espresso condition checks.